### ISSUE TITLE
BUG: Formatting MultiIndex crashes if first column is too long

### ISSUE THREAD

**Issue by futurulus** [pandas-dev/pandas#54695]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
print(pd.MultiIndex.from_product([['a'*58], ['b', 'c']]))
```

### Issue Description

Raises an exception in formatting:

```
Traceback (most recent call last):
  File "/Users/monroe/Code/lexicon/grammar_strength/pandas_bug_1.py", line 2, in <module>
    print(pd.MultiIndex.from_product([['a'*58], ['b', 'c']]))
  File "/Users/monroe/.pyenv/versions/pyenv-39/lib/python3.9/site-packages/pandas/core/indexes/base.py", line 1233, in __repr__
    data = self._format_data()
  File "/Users/monroe/.pyenv/versions/pyenv-39/lib/python3.9/site-packages/pandas/core/indexes/base.py", line 1275, in _format_data
    return format_object_summary(
  File "/Users/monroe/.pyenv/versions/pyenv-39/lib/python3.9/site-packages/pandas/io/formats/printing.py", line 436, in format_object_summary
    word = tail_item + sep + " "
TypeError: can only concatenate tuple (not "str") to tuple
```

### Expected Behavior

Should print a string representation of the index, something like
```
MultiIndex([('aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa', 'b'),
            ('aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa', 'c')],
           )
```
(but possibly with truncation)

### Installed Versions

<details>

```
INSTALLED VERSIONS
------------------
commit           : 0f437949513225922d851e9581723d82120684a6
python           : 3.9.13.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 20.6.0
Version          : Darwin Kernel Version 20.6.0: Tue Jun 21 20:50:28 PDT 2022; root:xnu-7195.141.32~1/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 2.0.3
numpy            : 1.22.4
pytz             : 2022.2.1
dateutil         : 2.8.2
setuptools       : 58.1.0
pip              : 23.1.2
Cython           : None
pytest           : 6.1.2
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.9.2
html5lib         : None
pymysql          : None
psycopg2         : 2.8.6
jinja2           : 3.1.2
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           : 1.0.9
fastparquet      : None
fsspec           : 2023.5.0
gcsfs            : None
matplotlib       : 3.5.3
numba            : 0.56.0
numexpr          : None
odfpy            : None
openpyxl         : 3.0.10
pandas_gbq       : None
pyarrow          : 12.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.9.1
snappy           : 
sqlalchemy       : 1.4.41
tables           : None
tabulate         : 0.9.0
xarray           : None
xlrd             : 2.0.1
zstandard        : None
tzdata           : 2023.3
qtpy             : None
pyqt5            : None
```

</details>

**Comment by rsm-23** [comment]
Seems to be working for me on `2.0.0rc0+1322.gaefede93b1`
```python
In [36]: pd.MultiIndex.from_product([['a'*58], ['b', 'c']])
Out[36]: 
MultiIndex([('aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa', ...),
            ('aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa', ...)],
           )
```

**Comment by futurulus** [comment]
@rsm-23 does this reproduce the error for you?

```python
import pandas as pd
with pd.option_context('display.width', 80):
    pd.MultiIndex.from_product([['a'*58], ['b', 'c']])
```

(80 is the default value of `display.width`, I think.)

I can see the error in 2.0.0 (from PyPI, not rc0) as well as 2.0.3, on Ubuntu (from a fresh install) and Mac, and in Jupyter as well as system terminal. It goes away if I set `display.width` to 100.

**Comment by mroeschke** [comment]
This seems to work on main so closing

```python
In [1]: import pandas as pd
   ...: print(pd.MultiIndex.from_product([['a'*58], ['b', 'c']]))
MultiIndex([('aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa', ...),
            ('aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa', ...)],
           )

In [2]: pd.__version__
Out[2]: '3.0.0.dev0+892.g41f3c2ecfa'
```

**Comment by futurulus** [comment]
Can confirm this is fixed in 2.1.0 and later. Thank you!!
